go-controller/pkg/ovn/gateway.go

```diff
@@ -249,14 +249,23 @@ func (gw *GatewayManager) GatewayInit(
 	hostSubnets []*net.IPNet,
 	l3GatewayConfig *util.L3GatewayConfig,
 	sctpSupport bool,
-	gwLRPIfAddrs, drLRPIfAddrs []*net.IPNet,
+	gwLRPJoinIPs, drLRPIfAddrs []*net.IPNet,
 	externalIPs []net.IP,
 	enableGatewayMTU bool,
 ) error {
 
 	gwLRPIPs := make([]net.IP, 0)
-	for _, gwLRPIfAddr := range gwLRPIfAddrs {
-		gwLRPIPs = append(gwLRPIPs, gwLRPIfAddr.IP)
+	for _, gwLRPJoinIP := range gwLRPJoinIPs {
+		gwLRPIPs = append(gwLRPIPs, gwLRPJoinIP.IP)
+	}
+	if gw.netInfo.TopologyType() == types.Layer2Topology {
+		// At layer2 GR LRP acts as the layer3 ovn_cluster_router so we need
+		// to configure here the .1 address, this will work only for IC with
+		// one node per zone, since ARPs for .1 will not go beyond local switch.
+		// This is being done to add the ICMP SNATs for .1 podSubnet that OVN GR generates
```

Comment:
This wasn't mentioned in the commit message and  I wasted a bunch of time trying to figure out what this had to do with lb-force-snat and services and this commit. In the future it would save the reviewer time if you broke it into a separate commit and better explanation as to why this was added. I think adding to  gwLRPIPs has more consequences, and it's not clear to me which are intended and unintended. I spawn a cluster with your patch and I see:
```
    nat 03fd7fe8-5764-4b49-b5f4-fb873440b142
        external ip: "169.254.0.12"
        logical ip: "50.0.0.0/16"
        type: "snat"
    nat 7037afdf-00aa-4a74-9df1-c8005ad9f5c4
        external ip: "169.254.0.11"
        logical ip: "50.0.0.0/16"
        type: "snat"
    nat 8659a29b-00f1-4b4f-b3f8-713565368c94
        external ip: "169.254.0.11"
        logical ip: "100.65.0.2"
        type: "snat"
    nat a05b30ed-f081-4db7-9362-8631507a6522
        external ip: "169.254.0.11"
        logical ip: "50.0.0.1"
        type: "snat"
```

was your intention here to add this final snat: 50.0.0.1 -> 169.254.0.11? To cover the ICMP case?

Reply:
If so, why would you add this for this PR? Is it because some service tests were failing without this? I thought we were going to submit another PR to remove this SNAT because OVN has fixed the issue.

Reply:
actually now I'm not sure at all because if i check a regular origin/master GR:

```
router d085b89d-3f55-4d00-9ef3-1dd735eeca94 (GR_default.l2.primary_ovn-control-plane)
    port rtoe-GR_default.l2.primary_ovn-control-plane
        mac: "02:42:ac:12:00:02"
        ipv6-lla: "fe80::42:acff:fe12:2"
        networks: ["169.254.0.11/17", "172.18.0.2/16"]
    port rtos-default.l2.primary_ovn_layer2_switch
        mac: "0a:58:64:41:00:02"
        ipv6-lla: "fe80::858:64ff:fe41:2"
        networks: ["10.20.0.1/16", "100.65.0.2/16"]
    nat 1eff8c60-69bb-4be0-8ed8-f27458bc9f5b
        external ip: "169.254.0.12"
        logical ip: "10.20.0.0/16"
        type: "snat"
    nat 6c1ce49b-ca7b-462c-981c-2001db59c84e
        external ip: "169.254.0.11"
        logical ip: "10.20.0.1"
        type: "snat"
    nat eacc64bd-f9d3-4c49-bf76-df232ad525bd
        external ip: "169.254.0.11"
        logical ip: "100.65.0.2"
        type: "snat"
```

it already has the .1 address...

Reply:
ok now I realize you were just moving around when you add the .1 IP in:
https://github.com/ovn-kubernetes/ovn-kubernetes/commit/dca22499b961a2ee0f03e524e43e82e767b79f7c#diff-cbf27b8cde9a56e399efde996bf12bc642fee29aae6c0bd8236a42105880ebdbL722-L728

but, still something seems weird that there is an extra nat with your patch

Reply:
ah it was cause I deployed without -ds